server/src/main/java/io/crate/analyze/relations/UnionSelect.java

```diff
@@ -53,15 +54,20 @@ public UnionSelect(AnalyzedRelation left, AnalyzedRelation right, boolean isDist
         this.right = right;
         this.name = new RelationName(null, UUIDs.dirtyUUID().toString());
         // SQL semantics dictate that UNION uses the column names from the first relation (top or left side)
-        List<Symbol> fieldsFromLeft = left.outputs();
-        ArrayList<ScopedSymbol> outputs = new ArrayList<>(fieldsFromLeft.size());
-        for (int i = 0; i < fieldsFromLeft.size(); i++) {
-            Symbol field = fieldsFromLeft.get(i);
-            Symbol rightField = right.outputs().get(i);
-            DataType<?> type = field.valueType().precedes(rightField.valueType())
-                ? field.valueType()
-                : rightField.valueType();
-            outputs.add(new ScopedSymbol(name, Symbols.pathFromSymbol(field), type));
+        var leftOutputs = left.outputs();
```

Comment:
imho, and since `UnionSelect` ctor is only used in `RelationAnalyzer#visitUnion()`, I'd keep all this processing in the RelationAnalyzer, and pass the union `ouput` as arg to the ctor. I find it a "cleaner" approach than doing validation in the ctor.

Reply:
Doing validations in the constructor has the advantage that it becomes impossible to create invalid objects. That it's only used in a single place might change